core/jobs/batch_jobs/state_interaction_validation_jobs.py

```diff
@@ -0,0 +1,1382 @@
+
+            lower_bound: int. The lower bound.
+
+            upper_bound: int. The upper bound.
+
+            lb_inclusive: bool. If lower bound is inclusive.
+
+            ub_inclusive: bool. If upper bound is inclusive.
+        """
+        range_var['lower_bound'] = lower_bound
+        range_var['upper_bound'] = upper_bound
+        range_var['lb_inclusive'] = lb_inclusive
+        range_var['ub_inclusive'] = ub_inclusive
+
+    @staticmethod
+    def _is_enclosed_by(range_var, range_ele) -> bool:
+        """Checks whether the ranges of rules enclosed or not
+
+        Args:
+            range_var: dict[str, Any]. To keep track of each rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
+
+            range_ele: dict[str, Any]. To keep track of other rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
+
+        Returns:
+            is_enclosed: bool. Returns True if both rule's ranges are enclosed.
+        """
+        if (
+            range_ele['lower_bound'] is None or range_var['lower_bound'] is None
+            or range_ele['upper_bound'] is None or (
+                range_var['upper_bound'] is None)
+        ):
+            return False
+
+        lb_satisfied = (
+            range_ele['lower_bound'] < range_var['lower_bound'] or
+            (range_ele['lower_bound'] == range_var['lower_bound'] and (
+                not range_var['lb_inclusive'] or range_ele['lb_inclusive'])
+            )
+        )
+
+        ub_satisfied = (
+            range_ele['upper_bound'] > range_var['upper_bound'] or
+            (range_ele['upper_bound'] == range_var['upper_bound'] and (
+                not range_var['ub_inclusive'] or range_ele['ub_inclusive'])
+            )
+        )
+
+        is_enclosed = lb_satisfied and ub_satisfied
+        return is_enclosed
+
+    @staticmethod
+    def _should_check_range_criteria(earlier_rule, later_rule) -> bool:
+        """Checks the range criteria between two rules by comparing their
+        rule type
+
+        Args:
+            earlier_rule: state_domain.RuleSpec. Previous rule.
+            later_rule: state_domain.RuleSpec. Current rule.
+
+        Returns:
+            bool. Returns True if the rules passes the range criteria check.
+        """
+        if (
+            (
+                earlier_rule.rule_type == 'IsExactlyEqualTo' and
+                later_rule.rule_type == 'IsExactlyEqualTo'
+            ) or
+            (
+                earlier_rule.rule_type == 'IsExactlyEqualTo' and
+                later_rule.rule_type == 'IsEquivalentTo'
+            ) or
+            (
+                earlier_rule.rule_type == 'IsExactlyEqualTo' and
+                later_rule.rule_type == 'IsEquivalentToAndInSimplestForm'
+            )
+        ):
+            return False
+        return True
```

Comment:
```suggestion
        if earlier_rule.rule_type != 'IsExactlyEqualTo':
            return True
        return later_rule.rule_type in ('IsExactlyEqualTo', 'IsEquivalentTo', 'IsEquivalentToAndInSimplestForm')
```

This should do the same thing, right?

Reply:
Yep!, updated